### PR TITLE
docs: adds forms documentation + demo

### PR DESCRIPTION
**Pull request by pixelflips**
# Description
Adds documentation and demo on using pds components within an HTML `<form>` element

Fixes https://kajabi.atlassian.net/browse/DSS-1149

## Type of change

Documentation

- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

### CODE REVIEW

File: libs/core/src/stories/forms.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    </pds-box>
+
+    <pds-box display="block" padding="sm">
+      <pds-switch component-id="notifications" label="Enable notifications"></pds-switch>
+    </pds-box>
+
+    <pds-box display="block" padding="sm">
+      <pds-textarea component-id="message" label="Message"></pds-textarea>
+    </pds-box>
+
```
Comment (pixelflips) [inline libs/core/src/stories/forms.docs.mdx:246]: TODO: Will need to add a select example once the component is available.

**Review by monicawheeler — APPROVED** [review]

**Review by ju-Skinner — APPROVED** [review]
Awesome job on this. 👏🏾  :shipit:

**Review by QuintonJason — APPROVED** [review]
Really nice 🏆

**Review by anechol — APPROVED** [review]
Very good!

**Comment by pixelflips** [comment]
Reminder: follow up with @FuturaExtraBold about adding select to examples once component is complete.

**Pull request merged**